src/core/producer/route/route_producer.cpp

```diff
@@ -143,7 +221,12 @@ class route_producer
             std::pair<core::draw_frame, core::draw_frame> frame;
             if (!buffer_.try_pop(frame)) {
                 graph_->set_tag(diagnostics::tag_severity::WARNING, "late-frame");
+                // If we have variable cadence or interlacing and are cross-channel, preserve the previous frame
+                if ((has_variable_cadence_ || is_interlaced_) && previous_frame_ && is_cross_channel_) {
```

Comment:
I don't follow why we need to store `previous_frame_` and not just whatever is already in `frame_`?

As this is written, if we end up here, that means the buffer has no frame, so frame X is 'missing'. frame_ currently contains X-1, previous_frame contains X-2. so why do we want to promote X-2 to be the current frame?

It is worth noting that both frame_ and previous_frame are pairs of 'frames' (ie a pair of fields)

-----

The result of this (and the line in the else), is that when routing a interlaced channel into a progressive channel, the second field gets stuck.